### PR TITLE
feat(api): Add auth/login endpoint

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Implements basic username / password auth via the API

### CODE REVIEW

File: src/sentry/api/endpoints/auth_login.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+from sentry.api.serializers.models.user import DetailedUserSerializer
+from sentry.web.forms.accounts import AuthenticationForm
+from sentry.web.frontend.base import OrganizationMixin
+
+
+class AuthLoginEndpoint(Endpoint, OrganizationMixin):
+    # Disable authentication and permission requirements.
+    permission_classes = []
+
+    def org_redirect_url(self, request):
```
Comment (wedamija) [inline src/sentry/api/endpoints/auth_login.py]: This is copy/pasted from `AuthConfigEndpoint.org_redirect_url`, can you DRY this up?

File: tests/sentry/api/endpoints/test_auth_login.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+from sentry.testutils import APITestCase
+
+
+class AuthLoginEndpointTest(APITestCase):
+    @fixture
+    def path(self):
+        return reverse('sentry-api-0-auth-login')
+
+    def setUp(self):
+        # Requets to set the test cookie
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_auth_login.py]: I don't think the new endpoint uses test cookies.

File: src/sentry/api/endpoints/auth_login.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        from sentry import features
+
+        # TODO(dcramer): deal with case when the user cannot create orgs
+        organization = self.get_active_organization(request)
+
+        if organization:
+            return organization.get_url()
+        if not features.has('organizations:create'):
+            return '/auth/login'
```
Comment (markstory) [inline src/sentry/api/endpoints/auth_login.py]: Should this have a trailing `/`?

File: tests/sentry/api/endpoints/test_auth_login.py (hunk 2 above)
Comment (EvanPurkhiser) [reply tests/sentry/api/endpoints/test_auth_login.py]: It does. In your original branch you removed it from the form, but I added it back and just made sure the config endpoint sets the test cookie.

**Review by wedamija — APPROVED** [review]

**Pull request merged**